content/blogs/gui-setup-part2-partitioning.md

```diff
@@ -0,0 +1,88 @@
+author:     "hbelusca"
+date:       2024-01-11
+tags:       [ newsletter, gui-setup ]
+---
+
+Greetings!
+
+As you may have noticed, I have been quite silent about my work, not writing on a regular basis blog entries about what I have done so far.
+Well, I am more concerned about getting _actual_ coding done and working before writing about it, instead of writing about half-done not-yet-tested code, much like what the GUI setup code was during these previous months.
+
+This is why it is only now that I'm presenting, in this second blog post, the work I made during the months of October and November, in preparation for the new 1st-stage ReactOS GUI setup.
+
+During the first week of October I finished addressing some last issues that arose with the partial setupapi winesync [I started back in September](/blogs/gui-setup-part1-setupapi).
+
+I then finally started the main part of this project: working on the GUI setup proper.
+The main necessary work for it resides in three functionalities:
+- disk partitioning and bootloader installation,
+- file copying,
+- settings saving in registry.
+While the last two features were already partly implemented (most of the code being shared with the existing text-mode installer ``USETUP'') and will be discussed in future blog posts, the first feature: disk partitioning, necessitated more work than I originally anticipated. Indeed, this functionality is shared with USETUP, but is used in a slightly different manner in the GUI setup.
+
+
+## Partitioning workflow - Text-mode versus GUI-mode setup
+
+# On the usual text-mode setup
+
+Long-time ReactOS users (and Windows XP/2k3 alike) certainly remember that, during the 1st-stage _text-mode_ setup (hereafter called ``USETUP''), disk partitions can be created, deleted, and selected from the partition list screen.
+
+<<< IMAGE PARTITION LIST USETUP >>>
+
+While the creation/deletion operations made from this screen are not applied immediately to the disk, they get applied _as soon_ as the user selects a suitable (created or existing) partition, or an eligible empty space, for installing ReactOS.
+There, the new partition layout is written to the disk, then installation continues with the selection of a filesystem for formatting newly-created partitions, and for the selected one as well.
+
+<<< IMAGE FILESYSTEM SELECTION >>>
+
+Once a suitable filesystem is chosen, the actual formatting of the partition takes place.
+After all the newly-created partitions (and perhaps the selected one) have been formatted, these new partitions are checked for defect with CHKDSK.
+This is particularly true for the selected ([_boot_](http://homepage.ntlworld.com/jonathan.deboynepollard/FGA/boot-and-system-volumes.html)) partition for ReactOS installation (as well as the [_system_](https://en.wikipedia.org/wiki/System_partition_and_boot_partition) partition where the bootloader gets installed; usually on BIOS-based PCs these can be the same, but can otherwise be different).
```

Comment:
It is a bit weird that you link to two completely different pages (and the first one does not load for me)

Reply:
yes the URL is an old one I had referenced somewhere; I'll have to find an updated one.